banzai/main.py

```diff
@@ -241,126 +238,55 @@ def reduce_night():
-
-    for stage in stages_to_do:
-        stage_to_run = stage(pipeline_context)
-        images = stage_to_run.run(images)
-
-    output_files = image_utils.save_images(pipeline_context, images,
-                                           master_calibration=calibration_maker)
-    return output_files
+def get_preview_stages_todo(image_suffix):
+    if image_suffix == 'b00.fits':
+        stages = get_stages_todo(last_stage=trim.Trimmer,
+                                 extra_stages=[bias.BiasMasterLevelSubtractor, bias.BiasComparer])
+    elif image_suffix == 'd00.fits':
+        stages = get_stages_todo(last_stage=bias.BiasSubtractor,
+                                 extra_stages=[dark.DarkNormalizer, dark.DarkComparer])
+    elif image_suffix == 'f00.fits':
+        stages = get_stages_todo(last_stage=dark.DarkSubtractor,
+                                 extra_stages=[flats.FlatNormalizer, qc.PatternNoiseDetector, flats.FlatComparer])
+    else:
+        stages = get_stages_todo()
+    return stages
 
 
 def run_preview_pipeline():
     parser = argparse.ArgumentParser(
         description='Make master calibration frames from LCOGT imaging data.')
-
-    parser.add_argument("--processed-path", default='/archive/engineering',
-                        help='Top level directory where the processed data will be stored')
-    parser.add_argument("--log-level", default='debug', choices=['debug', 'info', 'warning',
-                                                                 'critical', 'fatal', 'error'])
-    parser.add_argument('--post-to-archive', dest='post_to_archive', action='store_true',
-                        default=False)
-    parser.add_argument('--post-to-elasticsearch', dest='post_to_elasticsearch', action='store_true',
-                        default=False)
-    parser.add_argument('--db-address', dest='db_address',
-                        default='mysql://cmccully:password@localhost/test',
-                        help='Database address: Should be in SQLAlchemy form')
-    parser.add_argument('--fpack', dest='fpack', action='store_true', default=False,
-                        help='Fpack the output files?')
-    parser.add_argument('--rlevel', dest='rlevel', default=11, help='Reduction level')
     parser.add_argument('--n-processes', dest='n_processes', default=12,
                         help='Number of listener processes to spawn.', type=int)
     parser.add_argument('--broker-url', dest='broker_url',
                         default='amqp://guest:guest@rabbitmq.lco.gtn:5672/',
                         help='URL for the broker service.')
     parser.add_argument('--queue-name', dest='queue_name', default='preview_pipeline',
                         help='Name of the queue to listen to from the fits exchange.')
-    parser.add_argument('--max-preview-tries', dest='max_preview_tries', default=5,
-                        help='Maximum number of tries to produce a preview image.')
```

Comment:
I think this should stay as a preview pipeline kwarg

Reply:
I mostly agree with that. Except this is looking forward to that is the main type of reduction.

Reply:
Ah oops, I missed that `max_preview_tries` is now just `max_tries`